homeassistant/components/habitica/sensor.py

```diff
@@ -263,25 +206,27 @@ def __init__(
             identifiers={(DOMAIN, entry.unique_id)},
         )
 
-    async def async_update(self) -> None:
-        """Update Sensor state."""
-        await self.coordinator.update()
-        data = self.coordinator.data
+    @property
+    def native_value(self) -> StateType:
+        """Return the state of the device."""
+        data = self.coordinator.data.user
         for element in self.entity_description.value_path:
             data = data[element]
-        self._attr_native_value = data
+        return data  # type: ignore[return-value]
```

Comment:
Why the ignore?

Reply:
linter complains that return value cannot be a dict

Reply:
But how is it typed as a dict?

Reply:
the return value here is a dict[str, Any], the for loop then does a recursive search for the value in a nested dict
`data = self.coordinator.data.user
`

Reply:
Should we cast it then?

Reply:
this should do the trick ^^
`cast(StateType, data)`